anesthetic/utils.py

```diff
@@ -55,10 +55,11 @@ def quantile(a, q, w):
 
 def check_bounds(d, xmin=None, xmax=None):
     """Check if we need to apply strict bounds."""
-    if xmin is not None and (d.min() - xmin)/(d.max()-d.min()) > 1e-2:
-        xmin = None
-    if xmax is not None and (xmax - d.max())/(d.max()-d.min()) > 1e-2:
-        xmax = None
+    if len(d) > 0:
+        if xmin is not None and (min(d) - xmin)/(max(d)-min(d)) > 1e-2:
+            xmin = None
+        if xmax is not None and (xmax - max(d))/(max(d)-min(d)) > 1e-2:
+            xmax = None
```

Comment:
I'm getting this error now:
```python
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-26-3db102adf145> in <module>()
----> 1 fig, axes= pc_lcdm.plot_2d(['A_s', 'n_s', 'r'], label='LCDM')

/scratch/.virtualenvs/py27env/lib/python2.7/site-packages/anesthetic/samples.pyc in plot_2d(self, axes, *args, **kwargs)
    278                     ax_ = ax.twin if x == y else ax
    279                     self.plot(ax_, x, y, plot_type=types[pos], *args,
--> 280                               **local_kwargs[pos])
    281 
    282         return fig, axes

/scratch/.virtualenvs/py27env/lib/python2.7/site-packages/anesthetic/samples.pyc in plot(self, ax, paramname_x, paramname_y, *args, **kwargs)
    155 
    156             return plot(ax, x, y, xmin=xmin, xmax=xmax, ymin=ymin, ymax=ymax,
--> 157                         *args, **kwargs)
    158 
    159     def plot_1d(self, axes, *args, **kwargs):

/scratch/.virtualenvs/py27env/lib/python2.7/site-packages/anesthetic/plot.pyc in contour_plot_2d(ax, data_x, data_y, *args, **kwargs)
    351 
    352     x, y, pdf = kde_2d(data_x, data_y,
--> 353                        xmin=xmin, xmax=xmax, ymin=ymin, ymax=ymax)
    354     pdf /= pdf.max()
    355     p = sorted(pdf.flatten())

/scratch/.virtualenvs/py27env/lib/python2.7/site-packages/anesthetic/kde.pyc in kde_2d(d_x, d_y, xmin, xmax, ymin, ymax)
     79     f = [xmax is None or xmin is None,
     80          ymax is None or ymin is None]
---> 81     d_x_, d_y_ = mirror_2d(d_x, d_y, xmin, xmax, ymin, ymax)
     82 
     83     with warnings.catch_warnings():

/scratch/.virtualenvs/py27env/lib/python2.7/site-packages/anesthetic/utils.pyc in mirror_2d(d_x_, d_y_, xmin, xmax, ymin, ymax)
     79 def mirror_2d(d_x_, d_y_, xmin=None, xmax=None, ymin=None, ymax=None):
     80     """If necessary apply reflecting boundary conditions."""
---> 81     d_x = d_x_.copy()
     82     d_y = d_y_.copy()
     83 

AttributeError: 'list' object has no attribute 'copy'
```

Reply:
How easy would it be to add a failing test which shows this error?

Reply:
Test added.

The test for `samples.plot_2d` didn't check for 1d lists as input.